File: OEM/Domain/PlannedOperations/PlannedOperationDto.cs
using System;

namespace OEMAPI.Domain.PlannedOperations
{
    public class PlannedOperationDto
    {
        public Guid Id { get; set; }
        public Guid OperatorPlanId { get; set; }
        public string ResourceId { get; set; }
        public string StaffId { get; set; }
        public DateTime PlannedStart { get; set; }
        public DateTime PlannedEnd { get; set; }
        public string OperationType { get; set; }
        public string Status { get; set; }
    }
}

File: OEM/src/dto/PlannedOperationDTO.ts
export interface PlannedOperationDTO {
  id: string;
  operationPlanId: string;
  resourceId: string;
  staffId: string;
  plannedStart: string; // ISO string
  plannedEnd: string;   // ISO string
  operationType: string;
  status: string;
}

export interface CreatePlannedOperationDTO {
  operationPlanId: string;
  resourceId: string;
  staffId: string;
  plannedStart: string;
  plannedEnd: string;
  operationType: string;
}

export interface UpdatePlannedOperationDTO {
  plannedStart?: string;
  plannedEnd?: string;
  status?: string;
}
